[DOCS-9956] Add Cloud SIEM OCSF doc (#28964)
DataDog/documentation: content/en/security/cloud_siem/open_cybersecurity_schema_framework.md

OCSF integration in Datadog's Cloud SIEM enables:

* **Simplified detection rules**: A unified attribute structure means detection logic can be written once and applied across multiple sources.
* **Streamlined investigations**: Analysts no longer need to remember source-specific formats because one schema enables a single-query triage across providers.
* **Cross-source correlation**: Detection logic can correlate events across disparate services (for example, phishing and privilege escalation).
* **Scalable integration maintenance**: OCSF allows consistent schema expectations, even as new data sources are added.

## OCSF model

To normalize your security data, OCSF remaps your data based on the following components:

1. [Data types, attributes, objects and arrays](#data-types-attributes-objects-and-arrays)
1. [Event classes and categories](#event-categories-and-classes)
1. [Profiles](#profiles)
1. [Extensions](#extensions)

### Data types, attributes, objects, and arrays

Data types, attributes, objects, and arrays are the main components of the OCSF model.

| Name | Description |
| ---- | ----------- |

> | Data types | Data types define the data elements as integers, string, floating point numbers, boolean values.  |

Reviewer: ```suggestion
| Data types | Data types define data elements as integers, strings, floating-point numbers, and boolean values.  |
```

| Attributes | Attributes are the building blocks of the framework. They are used to provide the common language for your data, regardless of the source. See the [attribute dictionary][1] for a list of all attributes.  |
| Objects | Objects are collections of related attributes that represent the entities, such as a process, device, user, malware, or file.  |
| Arrays | Arrays support any of the data types, including complex types.  |

### Event categories and classes

Security events within the OCSF model are organized into categories, which are high-level groupings that sort events based on their data type. See [OCSF Categories][2] for more information and a list of available categories. Categories are further divided up into event classes. For example, there are [six classes][3] for the Identity & Access Management category. See [OCSF Event Classes][4] for more information.

### Profiles

> Profiles are a class of attributes that you can optionally overlay on event classes and objects that reference the classes. It adds additional information to an existing event class and is independent of event categories. See [OCSF Profiles][5] for a list of profiles and the [OCSF Profiles documentation][6] for more information.

Reviewer: ```suggestion
Profiles are a class of attributes that you can optionally overlay onto event classes and the objects that reference them. It adds additional information to an existing event class and is independent of event categories. See [OCSF Profiles][5] for a list of profiles and the [OCSF Profiles documentation][6] for more information.
```

### Extensions

You can optionally add extensions, such as new attributes, objects, categories, profiles, and event classes, to the OCSF schemas. See [OCSF Extensions][7] for more information.

## Supported out-of-the-box OCSF pipelines

The following Log Management integrations support out-of-the-box OCSF pipelines:

- Okta

> - Cloudtrail

Author: ```suggestion
- CloudTrail
```

- GitHub

## View Security Pipelines \- OCSF

> Cloud SIEM OCSF remaps log data in Log Management's [integration pipelines][8]. See [Support out-of-the-box OCSF pipelines](#supported-out-of-the-box-ocsf-pipelines) for details.

Reviewer: ```suggestion
Cloud SIEM OCSF remaps log data in Log Management's [integration pipelines][8]. See [Supported out-of-the-box OCSF pipelines](#supported-out-of-the-box-ocsf-pipelines) for details.
```

> To see the Integration Pipeline Library for a source:

Reviewer: ```suggestion
To view the Integration Pipeline Library for a source:
```

1. Navigate to [Logs Pipelines][9].
1. Click **Browse Pipeline Library**.
1. Search and click on the integration you are interested in (for example, Okta).

> 1. To see the OCSF pipelines for Okta, scroll down to the end of the list of processors for the Okta integration.

Reviewer: ```suggestion
1. To view the OCSF pipelines for Okta, scroll down to the end of the list of processors for the Okta integration.
```

To view the read-only OCSF pipeline for a source integration:
1. Navigate to [Logs Pipelines][9].
1. Select your pipeline.
1. Scroll to the OCSF pipelines at end of the pipeline's processors.

> 1. Click the OCSF pipeline to see the associated remap processors.

Reviewer: ```suggestion
1. Click the OCSF pipeline to view the associated remap processors.
```

> 1. Click the eye icon on the OCSF pipeline to see information such as the:

Reviewer: ```suggestion
1. Click the eye icon on the OCSF pipeline to view information such as the following:
```

    - OCSF schema version
    - Class
    - Profile

> **Note**: If you clone the main pipeline, the OCSF pipelines are log pipelines and not Security pipelines in the cloned version.

Reviewer: ```suggestion
**Note**: Cloning the main pipeline converts OCSF pipelines into log pipelines rather than Security pipelines.
```
